rcon/squad_automod/automod.py

```diff
@@ -219,80 +222,59 @@ def get_punitions_to_apply(rcon, config: NoLeaderConfig) -> PunitionsToApply:
                 
                 logger.info("Squad %s - %s doesn't have leader", team, squad_name)
 
-                state = should_warn_squad(watch_status, config, team, squad_name)
+                state_added = False
+                for player in squad["players"]:
+                    a_player = APlayer(
+                        steam_id_64=player["steam_id_64"],
+                        player=player["name"],
+                        team=team,
+                        squad=squad_name,
+                        role=player.get("role"),
+                        lvl=player.get("level"),
+                    )
 
-                if state == PunishStepState.apply:
-                    punitions_to_apply.warning[team].append(squad_name)
-                    punitions_to_apply.add_squad_state(squad_name, squad)
-                if state == PunishStepState.wait:
-                    punitions_to_apply.pending_warnings[team].append(squad_name)
-                    punitions_to_apply.add_squad_state(squad_name, squad)
-                if state != PunishStepState.go_to_next_step and state != PunishStepState.disabled:
-                    continue
+                    state = should_warn_player(watch_status, config, squad_name, player)
+
+                    if state == PunishStepState.apply:
+                        punitions_to_apply.warning.append(a_player)
+                        if not state_added:
+                            punitions_to_apply.add_squad_state(squad_name, squad)
+                            state_added = True
+                        continue
+
+                    if state == PunishStepState.wait:
+                        punitions_to_apply.pending_warnings.append(a_player)
```

Comment:
We could remove that now. I only added the pending warning to have the full picture everytime there's a broadcast. But if we're not using broadcasts anymore it's irrelevant

Reply:
I thought so to and removed it. At some point it made sense to me to still have it. I'll revisit that part and either give you a response or remove it 😅